script/deploy-core.s.sol

```diff
@@ -45,44 +65,92 @@ contract CoreDeployer is Script {
-            LBQuoter quoter = new LBQuoter(
-                deployment.factoryV1,
-                address(factoryV2),
-                address(routerV2)
-            );
-            console.log("LBQuoter deployed -->", address(quoter));
-
-            factoryV2.setLBPairImplementation(address(pairImplementation));
-            console.log("LBPair implementation set on factoryV2\n");
-
-            uint256 quoteAssets = ILBLegacyFactory(deployment.factoryV2).getNumberOfQuoteAssets();
-            for (uint256 j = 0; j < quoteAssets; j++) {
-                IERC20 quoteAsset = ILBLegacyFactory(deployment.factoryV2).getQuoteAsset(j);
-                factoryV2.addQuoteAsset(quoteAsset);
+            if(deployment.factoryV2 == address(0)) {
+                console.log("Deploying factory v2...");
+                factoryV2 = Upgrades.deployTransparentProxy(
+                    "LBFactory.sol",
+                    deployment.owner,
+                    ""
+                );
+
+                console.log("LBFactory deployed -->", factoryV2);
+            } else {
+                factoryV2 = deployment.factoryV2;
+            }
+
+            deployment.factoryV2 = factoryV2;
+
+            if(deployment.lbPairImplementation == address(0)) {
+                console.log("Deploying lbPairImplementation...");
+                lbPairImplementation = address(new LBPair(ILBFactory(factoryV2)));
+                console.log("lbPairImplementation deployed -->", lbPairImplementation);
+            } else {
+                lbPairImplementation = deployment.lbPairImplementation;
+            }
+
+            deployment.lbPairImplementation = lbPairImplementation;
+
+            
+            if(deployment.lbPairUpgradeableBeacon == address(0)) {
+                console.log("Deploying lbPairUpgradeableBeacon...");
+                lbPairUpgradeableBeacon = address(new LBPairUpgradeableBeacon(lbPairImplementation, deployer, factoryV2));
+                console.log("lbPairUpgradeableBeacon deployed -->", lbPairUpgradeableBeacon);
+            } else {
+                lbPairUpgradeableBeacon = deployment.lbPairUpgradeableBeacon;
+            }
+
+            deployment.lbPairUpgradeableBeacon = lbPairUpgradeableBeacon;
+
+            console.log("initializing lbFactory...");
+            LBFactory(factoryV2).initialize(deployer, deployer, FLASHLOAN_FEE, lbPairUpgradeableBeacon);
+            console.log("LBFactory initialized, owner --> ", LBFactory(factoryV2).owner());
+
+            if(deployment.routerV2 == address(0)) {
+                Options memory opts;
+                opts.constructorData = abi.encode(LBFactory(factoryV2), ISovrynLBFactoryV1(deployment.factoryV1), IWNATIVE(deployment.wNative));
+                console.log("Deploying routerV2...");
+                routerV2 = Upgrades.deployTransparentProxy(
+                    "LBRouter.sol",
+                    deployment.owner,
+                    "",
```

Comment:
LBRouter requires constructor parameters but none passed - the third param is an empty string